runtime/v1/linux/proc/exec.go

```diff
@@ -95,6 +95,8 @@ func (e *execProcess) setExited(status int) {
 
 func (e *execProcess) delete(ctx context.Context) error {
 	e.wg.Wait()
+	e.mu.Lock()
```

Comment:
what if we removed it from here and just applied the lock around the `s.transition()` calls in the Delete methods?

Reply:
Yes, `s.transition("deleted")` need apply lock, but I just wonder whether `delete` need a lock. And update as you said.